Implements PUT /api/resource_classes/ID to update the Flavors

marios opened the pull request:
This will delete all existing flavors and replace them with those in the
PUT body. Addresses https://github.com/tuskar/tuskar/issues/50

Testing please :)

If you run python sample_data.py - then you can test with:

```
PUT http://0.0.0.0:6385/v1/resource_classes/1

{
    "service_type": "foo",
    "name": "test-foo",
           "flavors": [
                { "name" : "foo",
                  "capacities" : [
                     {   "name": "cpu", 
                         "value" : "1",
                          "unit" : "count" }, 
                     {   "name": "memory",
                         "value" : "1",
                         "unit" : "MiB" },
                     {   "name": "storage", 
                         "value" : "1",
                         "unit" : "GiB" }
                  ]
                }
           ]
}
```

jistr commented:
Works for me. I have two notes:
- Please use comments instead of docstrings in the method body. Docstrings should be just the first statement in module/class/function. (See http://stackoverflow.com/questions/708649/python-comments-vs-strings)
- You added `as e` to the exception catch clause, which generates one more pep8 error (unused variable). Please revert that.

Conditional ACK wrt ^^

marios commented:
@jistr thanks for review - I will fix those and merge

EDIT - also thanks for the comments vs docstring education - i did not know that!

marios committed:
Implements PUT /api/resource_classes/ID to update the Flavors

This will delete all existing flavors and replace them with those in the
PUT body. Addresses https://github.com/tuskar/tuskar/issues/50
